pkg/asset/manifests/dns.go

```diff
@@ -244,31 +256,44 @@ func GetProjectForDNSZones(installConfig *installconfig.InstallConfig) string {
 // GetGCPPrivateZoneName attempts to find the name of the private zone for GCP installs. When a shared vpc install
 // occurs, a precreated zone may be used. If a zone is found (in this instance), then the zone should be paired with
 // the network that is supplied through the install config (when applicable).
-func GetGCPPrivateZoneName(ctx context.Context, client *icgcp.Client, installConfig *installconfig.InstallConfig, clusterID string) (string, error) {
-	privateZoneID := fmt.Sprintf("%s-private-zone", clusterID)
+func GetGCPPrivateZoneName(ctx context.Context, client *icgcp.Client, installConfig *installconfig.InstallConfig, clusterID string) (string, bool, error) {
+	defaultPrivateZoneID := fmt.Sprintf("%s-private-zone", clusterID)
+	privateZoneID := defaultPrivateZoneID
+	shouldCreateZone := true
+
 	if installConfig.Config.GCP.NetworkProjectID != "" {
 		project := GetProjectForDNSZones(installConfig)
 		if installConfig.Config.GCP.PrivateZone != nil && installConfig.Config.GCP.PrivateZone.Zone != "" {
+			// Override the default with the name provided. If this zone does not exist, then
+			// this should still be returned.
 			privateZoneID = installConfig.Config.GCP.PrivateZone.Zone
 		}
 
 		zone, err := client.GetDNSZone(ctx, project, installConfig.Config.ClusterDomain(), false)
 		if err != nil {
-			return "", fmt.Errorf("failed to get private zone for %q: %w", installConfig.Config.BaseDomain, err)
+			// CORS-4012: The user may specify a zone to be created if it does not exist.
+			// Do not fail if the specified zone does not exist.
+			if privateZoneID != defaultPrivateZoneID {
```

Comment:
`defaultPrivateZoneID` has the format `%s-private-zone`. `privateZoneID` is obtained from installconfig. Do we know it would be the same format before this check here? If the user needs to provide `installConfig.Config.GCP.PrivateZone.Zone` in the format `%s-private-zone`, we should add it to the doc string within installconfig. If not, should we append it with `private-zone`?